File: 백준/Silver/1463. 1로 만들기/1로 만들기.py
N = int(input())

dp = [0] * (N+1)
if N >= 2:
    dp[2] = 1

# N이 3 이상일 때 초기화
if N >= 3:
    dp[3] = 1
if N >= 4:
    for i in range(4,N+1):        
        dp[i] = dp[i-1] +1
        if i %2 ==0:
            dp[i] = min(dp[i],dp[i//2]+1)
        if i %3 ==0:
            dp[i] = min(dp[i],dp[i//3]+1)
print(dp[N])

File: 백준/Silver/1463. 1로 만들기/1로 만들기.swift
//
//  main.swift
//  Swiftalgorithm
//
//  Created by 김건호 on 9/10/24.
//

import Foundation

let n = Int(readLine()!)!

if n == 1 {
    print(0)
} else {
    var dp = Array(repeating: 0, count: n+1)

    for i in 2...n {
        dp[i] = dp[i-1] + 1
        
        if i % 2 == 0 {
            dp[i] = min(dp[i], dp[i/2] + 1)
        }
        if i % 3 == 0 {
            dp[i] = min(dp[i], dp[i/3] + 1)
        }
    }

    print(dp[n])
}
